### PR TITLE
[1.11] Add biome dictionary types for newer biomes

### PR DESCRIPTION
**Pull request by mezz**
The newer biomes had no types set in the `BiomeDictionary`, this PR adds them.
The formatting of this section is pretty annoying to update, hopefully nobody creates any more long biome names. (looking at you `MUTATED_EXTREME_HILLS_WITH_TREES`)

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/BiomeDictionary.java [hunk 1]
```diff
@@ -406,45 +407,67 @@ private static boolean containsType(Set<Type> types, Type type)
...
+        addTypes(Biomes.SAVANNA,                          HOT,      SAVANNA,    PLAINS,   SPARSE           );
+        addTypes(Biomes.SAVANNA_PLATEAU,                  HOT,      SAVANNA,    PLAINS,   SPARSE           );
+        addTypes(Biomes.MESA,                             MESA,     SANDY                                  );
+        addTypes(Biomes.MESA_ROCK,                        MESA,     SPARSE,     SANDY                      );
+        addTypes(Biomes.MESA_CLEAR_ROCK,                  MESA,     SANDY                                  );
+        addTypes(Biomes.VOID,                             VOID                                             );
+        addTypes(Biomes.MUTATED_PLAINS,                   PLAINS                                           );
+        addTypes(Biomes.MUTATED_DESERT,                   HOT,      DRY,        SANDY                      );
+        addTypes(Biomes.MUTATED_EXTREME_HILLS,            MOUNTAIN, SPARSE                                 );
+        addTypes(Biomes.MUTATED_FOREST, FOREST,           HILLS                                            );
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/common/BiomeDictionary.java]: `FOREST` needs lining up

File: src/main/java/net/minecraftforge/common/BiomeDictionary.java [hunk 2]
```diff
@@ -406,45 +407,67 @@ private static boolean containsType(Set<Type> types, Type type)
...
+        addTypes(Biomes.MUTATED_JUNGLE,                   HOT,      WET,        DENSE,    JUNGLE,  MOUNTAIN);
+        addTypes(Biomes.MUTATED_JUNGLE_EDGE,              HOT,      SPARSE,     JUNGLE,   HILLS            );
+        addTypes(Biomes.MUTATED_BIRCH_FOREST,             FOREST,   DENSE,      HILLS                      );
+        addTypes(Biomes.MUTATED_BIRCH_FOREST_HILLS,       FOREST,   DENSE,      MOUNTAIN                   );
+        addTypes(Biomes.MUTATED_ROOFED_FOREST,            SPOOKY,   DENSE,      FOREST,   MOUNTAIN         );
+        addTypes(Biomes.MUTATED_TAIGA_COLD,               COLD,     CONIFEROUS, FOREST,   SNOWY,   MOUNTAIN);
+        addTypes(Biomes.MUTATED_REDWOOD_TAIGA,            DENSE,    FOREST                                 );
+        addTypes(Biomes.MUTATED_REDWOOD_TAIGA_HILLS,      DENSE,    FOREST,     HILLS                      );
+        addTypes(Biomes.MUTATED_EXTREME_HILLS_WITH_TREES, MOUNTAIN, SPARSE                                 );
+        addTypes(Biomes.MUTATED_SAVANNA, MOUNTAIN,        HOT,      DRY,        SPARSE,   SAVANNA          );
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/common/BiomeDictionary.java]: As does `MOUNTAIN`

**Review by bs2609 — COMMENTED** [review]
I'd also say to make a `MUTATED` type, but I'll understand if you think that's too much effort

**Comment by mezz** [comment]
`MUTATED` biomes don't have a lot in common besides the fact that they're rare and most of them are more mountainous. I think the types here capture their properties reasonably well.

**Comment by bs2609** [comment]
Yeah, `RARE` would be a better property than `MUTATED`. The properties you have now are good though.

**Comment by mezz** [comment]
Added a new `RARE` type to the mutated biomes and a few of the regular ones.

**Comment by LexManos** [comment]
You dont have to indent everything else if there is one stupid thing with a stupidly long name.

**Comment by mezz** [comment]
Good to know. A lot of the new one were pretty long since they all have "mutated_" on top of existing names.

**Pull request merged**